R/server.R

```diff
@@ -12,51 +63,78 @@ server <- function(input, output, session, snapshot) {
     reset = reactive(input$reset)
   )
 
-  # Side Panel
-  add_metadata_to_sidebar(input, output, session, snapshot)
+  # ---
+  # Reveal filters based on tab
+  observeEvent(input$primary_nav_bar, {
+    if (input$primary_nav_bar == "Study Overview") {
+      shinyjs::hide("metric")
+      shinyjs::hide("site")
+      shinyjs::hide("participant")
+      updateSelectInput(session, "site", selected = "None")
+    } else if (input$primary_nav_bar == "Metric Details") {
+      shinyjs::show("metric")
+      shinyjs::show("site")
+      shinyjs::hide("participant")
+      updateSelectizeInput(session, "participant", selected = "None")
+    } else if (input$primary_nav_bar == "Participant Details") {
+      shinyjs::show("metric")
+      shinyjs::show("site")
+      shinyjs::show("participant")
+    }
+  })
 
   # ----
   # Study
   # add_metric_observer(snapshot, reactives$metric)
-  study_overview_server("study_overview", snapshot)
-  modScatterServer("scatter", snapshot, reactives$site)
-
-  # ----
-  # Metric
-  initialize_metric_select(input, output, session, snapshot)
-  observe_metric_select(snapshot, reactives$metric)
-  metric_details_server(
-    "metric_details",
-    snapshot,
-    reactives$metric,
-    reactives$site
+  study_overview_server("study_overview",
+    dfResults = dfResults,
+    dfGroups = dfGroups,
+    dfMetrics = dfMetrics
   )
 
-  # ----
-  # Site
-  initialize_site_select(input, output, session, snapshot)
-  observe_site_select(reactives$site, snapshot)
-  site_details_server(
-    "site_details",
-    snapshot,
-    reactives$site,
-    reactives$metric
+  modScatterServer("scatter",
```

Comment:
Let's be consistent about naming. `study_overview_server()` and `modScatterServer()` are using different naming schemes. Their filenames follow the same pattern, but not the function names. Most functions appear to follow the (somewhat more standard for R) `snake_case` format, so let's standardize around that (without the `mod_` piece, too, since the others don't use it). So this should be `scatter_server`, and should be updated in its definition in `mod_scatter_server.R` as well.

Reply:
Are you wanting that done on this PR or will this be a ticket?

Reply:
I went ahead and make a ticket for this. I don't recall how `modScatterServer` made its way in there. There were different developers over time, so some of the naming conventions were adopted inconsistently.

Reply:
#152